### PR TITLE
Improvements of CloudMaskTask

### PR DESCRIPTION
**Pull request by AleksMat**
Changes:
- Added unit tests for multi-temporal cloud masking which were currently missing.
- Converted lightgbm models into their native format and updated in the code.
- Updated subpackage dependencies - it turns out neither s2cloudless nor scikit-learn are required for eo-learn-mask anymore.
- Various code-style improvements in the task. More can be done in the future but the task is so complex that many things cannot be much simplified anyway.

In the end, I was a bit annoyed with `self.sigma` because the fact that it is set only in the execute method makes the task unsafe for multi-threaded execution in a workflow. However, changing this would make the code much messier because sigma parameter would have to be passed almost everywhere. Since running this task in a multi-threaded way isn't very efficient anyway I decided to just leave it for now. But I'll open an internal issue about that.

### CODE REVIEW

**Review by batic — COMMENTED** [review]
Looks ok, provided data and masks used in tests were competed before the refactoring :)

File: mask/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 eo-learn-core
-s2cloudless>=1.5.0
+numpy
+sentinelhub
 opencv-contrib-python-headless
-joblib
-scikit-learn>=0.19.0,<0.24
+scikit-image>=0.13.0
```
[the remark this answers is not in the document]
Comment (AleksMat) [reply mask/requirements.txt:5]: This is `scikit-image` and not `scikit-learn`. So version `0.19.0` doesn't exist yet.

Before `scikit-image` was a dependency of `s2cloudless` so we didn't have it on the list here. But now I had to add it.

File: mask/eolearn/tests/test_cloud_mask.py [hunk 2]
```diff
@@ -45,28 +45,27 @@ def _check_shape(self, output, data):
...
 
-    def test_cloud_coverage(self):
-        # Classifier is run on same resolution as data array
+    def test_mono_temporal_cloud_detection(self):
         add_tcm = CloudMaskTask(data_feature='ALL_DATA',
                                 all_bands=True,
                                 is_data_feature='IS_DATA',
                                 mono_features=('CLP_TEST', 'CLM_TEST'),
-                                mask_feature=None,
+                                # multi_features=('CLP_TEST_MULTI', 'CLM_TEST_MULTI'),
```
[the remark this answers is not in the document]
Comment (AleksMat) [reply mask/eolearn/tests/test_cloud_mask.py]: Thanks, I forgot to remove this. It's fixed now

**Review by batic — APPROVED** [review]
